arch/x86/common/source/IDEDriver.cpp

```diff
@@ -88,77 +80,83 @@ uint32 IDEDriver::doDeviceDetection()
       else
       {
         outportbp(base_port + 6, (cs % 2 == 0 ? 0xA0 : 0xB0));
+        outportbp(base_port + 7, 0xEC); // IDENTIFY
 
-        uint8 c1 = inportbp(base_port + 2);
-        uint8 c2 = inportbp(base_port + 3);
+        uint8 sr = inportbp(base_port + 7);
 
-        if (c1 != 0x01 && c2 != 0x01)
-          debug(IDE_DRIVER, "doDetection: Not found after reset ! \n");
```

Comment:
So this is what seemed strange to me.
It selects the device and then checks offset 2 (c1) and offset 3 (c2), which are supposed to be "Sector Count Register" and "Sector Number Register / (LBAlo)", to be 1.
What I thought could be the problem is that the device selection is not enough to guaranty the registers to be valid, since it seems like ATA_PIO can be a bit finicky.

The current fix is a bit of a workaround and I lazily just removed checking c1 and c2. They should probably be asserted to be 1, no?
It might not be necessary to issue the identify command to detect the device, but it seems like that makes sense. However, the identify command is now being send twice, because the ATADriver ctor also sends it.
Also I am aware this is just further adding to the spagettiness of this code...